packages/core/src/components/forms/numericInput.tsx

```diff
@@ -594,4 +668,8 @@ export class NumericInput extends AbstractPureComponent<HTMLInputProps & INumeri
         const scaleFactor = Math.pow(10, this.state.stepMaxPrecision);
         return Math.round(value * scaleFactor) / scaleFactor;
     }
+
+    private updateDelta(value: number) {
+        this.delta = value;
```

Comment:
🙅‍♂️ i meant that the entire logic should be here, _that's_ the part that shouldn't be repeated: 
```ts
// needs event arg
this.delta = this.getIncrementDelta(direction, e.shiftKey, e.altKey);
return this.delta;
```

Reply:
Gotcha.